jpa/eclipselink.jpa.test/src/org/eclipse/persistence/testing/tests/jpa/beanvalidation/BeanValidationJunitTest.java

```diff
@@ -342,6 +343,32 @@ public void testTraversableResolverPreventsLoadingOfLazyRelationships() {
         assertTrue( "Lazy field should not be instantiated because of validation", !isInstantiated(employee, "projects", project) );
         assertTrue( "Lazy field should not be instantiated because of validation", !isInstantiated(employee, "managedProject", project) );
     }
+    
+    public void testPesistWithInvalidAndModifiedBeanData() {
+        
+        EntityManager em = createEntityManager();
+        boolean gotConstraintViolations = false;
+        String invalidName = getFilledStringOfLength(Employee.NAME_MAX_SIZSE + 1);
+        String validName = getFilledStringOfLength(Employee.NAME_MAX_SIZSE - 1);
+        
+        // Persist an object with valid value
+        beginTransaction(em);
+        Employee e1 = new Employee(100, validName, validName, 1337);
+        em.persist(e1);
+
+        // (b) now we demonstrate how eclipse link fails to invoke the JSR validations on entity
+        // when we modify the entity during the transaction with invalid value
+        try {
+            e1.setName(invalidName);
+            e1.setSurname(invalidName);
+            em.persist(e1);
```

Comment:
No ```em.merge(e1)```? The entity is already managed at this point. I ran a quick local test and I don't see EclipseLink calling @PrePersist twice when there are multiple persists for the same managed entity. I can verify that with EclipseLink's implementation as well. Take a look at `org.eclipse.persistence.internal.sessions.UnitOfWorkImpl.registerNotRegisteredNewObjectForPersist`:

``` java
   /**
     * INTERNAL:
     * Called only by registerNewObjectForPersist method,
     * and only if newObject is not already registered.
     * Could be overridden in subclasses.
     */
```
If the object being persisted is already registered, we don't execute the PrePersistEvent.